From 3950979b8f49a743ac976551b16284340539d6df Mon Sep 17 00:00:00 2001
From: Alexandre Iooss <erdnaxe@crans.org>
Date: Fri, 27 Aug 2021 16:47:24 +0200
Subject: [PATCH] build: Define HAVE_RPM, HAVE_DPKG and HAVE_PACMAN

When using option `--with-distro`, `HAVE_RPM`, `HAVE_DPKG` and
`HAVE_PACMAN` are not defined and make the configure phase fail.
This makes sure that these conditionals are always defined.

(cherry picked from commit 210959cc344d6a4a1e3afa26d276b130651def74)
---
 m4/guestfs-appliance.m4 | 18 +++++++++---------
 1 file changed, 9 insertions(+), 9 deletions(-)

diff --git a/m4/guestfs-appliance.m4 b/m4/guestfs-appliance.m4
index 8b9ddcb5d..66aea5f98 100644
--- a/m4/guestfs-appliance.m4
+++ b/m4/guestfs-appliance.m4
@@ -114,16 +114,16 @@ AC_ARG_WITH([distro],
             AC_MSG_ERROR([/etc/os-release not available, please specify the distro using --with-distro=DISTRO])
         fi
     ]
-    AM_CONDITIONAL([HAVE_RPM],
-        [AS_CASE([$DISTRO], [REDHAT | SUSE | OPENMANDRIVA | MAGEIA ], [true],
-                            [*], [false])])
-    AM_CONDITIONAL([HAVE_DPKG],
-        [AS_CASE([$DISTRO], [DEBIAN | UBUNTU ], [true],
-                            [*], [false])])
-    AM_CONDITIONAL([HAVE_PACMAN],
-        [AS_CASE([$DISTRO], [ARCHLINUX | FRUGALWARE ], [true],
-                            [*], [false])])
 )
+AM_CONDITIONAL([HAVE_RPM],
+    [AS_CASE([$DISTRO], [REDHAT | SUSE | OPENMANDRIVA | MAGEIA ], [true],
+                        [*], [false])])
+AM_CONDITIONAL([HAVE_DPKG],
+    [AS_CASE([$DISTRO], [DEBIAN | UBUNTU ], [true],
+                        [*], [false])])
+AM_CONDITIONAL([HAVE_PACMAN],
+    [AS_CASE([$DISTRO], [ARCHLINUX | FRUGALWARE ], [true],
+                        [*], [false])])
 AC_SUBST([DISTRO])
 
 dnl Add extra packages to the appliance.
